src/app/clusters/color-control-server/color-control-server.cpp

```diff
@@ -2475,13 +2507,14 @@ void ColorControlServer::levelControlColorTempChangeCommand(EndpointId endpoint)
             currentLevel.SetNonNull((uint8_t) 0x7F);
         }
 
-        uint16_t tempPhysMax = MAX_TEMPERATURE_VALUE;
-        Attributes::ColorTempPhysicalMaxMireds::Get(endpoint, &tempPhysMax);
+        uint32_t tempCoupleMin = getTemperatureCoupleToLevelMin(endpoint);
+        uint32_t tempPhysMax   = MAX_TEMPERATURE_VALUE;
+        Attributes::ColorTempPhysicalMaxMireds::Get(endpoint, reinterpret_cast<uint16_t *>(&tempPhysMax));
```

Comment:
This reinterpret_cast is not ok.  Starting with the fact that it will break on a big-endian CPU.

The type-safe attribute getters are type-safe for a reason.  Please get into a `uint16_t`, and then if we want to assign into a `uint32_t` we can do that.

Reply:
Ok yes, I can see that.